src/redux/selectedThemeSlice.ts

```diff
@@ -0,0 +1,38 @@
+import { Theme, createTheme } from '@mui/material';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { themes } from '../utils/themes';
+
+const OSTheme = window
+  .matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+
+localStorage.setItem('OSTheme', OSTheme);
+
+const getTheme = (themeName: string): Theme => {
+  const theme = themes.find((foundTheme) => foundTheme.themeName === themeName)
+    || { themeName: '', themeObject: createTheme() };
+
+  return createTheme(theme.themeObject);
+};
+
+const themeNameFromLocalStorage = localStorage.getItem('theme');
+
+const initialState: Theme = themeNameFromLocalStorage
+  ? getTheme(themeNameFromLocalStorage)
+  : getTheme(OSTheme);
+
+const selectedThemeSlice = createSlice({
+  name: 'selectedTheme',
+  initialState,
+  reducers: {
+    selectTheme: (_theme, action: PayloadAction<string>) => {
```

Comment:
what does that _theme means?

Reply:
its a simple syntax, which means what this variable is not used anywhere

Reply:
i use it because linter complains that the variable not used anywhere